web/src/views/FileBrowserView/PublishFileBrowser.vue

```diff
@@ -194,6 +253,33 @@ export default {
         this.location = `${this.location}${name}`;
       }
     },
+
+    showDelete(item) {
+      const me = publishRest.user ? publishRest.user.username : null;
+
+      const isAdmin = me && publishRest.user.admin;
+      const isOwner = me && this.owners.includes(me);
+
+      return !item.folder && (isAdmin || isOwner);
+    },
```

Comment:
This is a small thing, but I think the `me`, `isAdmin`, and `isOwner` parts should be moved to a computed value, since the admin/owner status won't change item to item. I think a `me` computed value and a simple `isOwner` computed value (which would encompass the current `isAdmin` and `isOwner` variables, since admins qualify as owners anyway) would be fine.

Reply:
This is a good idea, but will it work with the `publishRest` value? That's a reactive store type of thing, but standalone. If I mention it in a computed prop, will the prop correctly update when `publishRest` changes?

Reply:
I'm pretty sure the only way this code as written will work is if `publishRest` is reactive. I can confirm that clicking `Logout` while browsing assets removes the delete icon, so I'm pretty sure this will work fine as a computed property.

Reply:
See da42a896b75e3d89ef4c19044fa2aa36bf14acc9.